core/services/pipeline/common_test.go

```diff
@@ -117,58 +127,200 @@ func Test_UnmarshalTaskFromMap(t *testing.T) {
 		require.Error(t, err)
 		require.Contains(t, err.Error(), "UnmarshalTaskFromMap: UnmarshalTaskFromMap only accepts a map[string]interface{} or a map[string]string")
 	})
+
+	t.Run("unknown task type", func(t *testing.T) {
+		t.Parallel()
+
+		taskMap := map[string]string{}
+		_, err := pipeline.UnmarshalTaskFromMap(pipeline.TaskType("xxx"), taskMap, 0, "foo-dot-id")
+		require.EqualError(t, err, `UnmarshalTaskFromMap: unknown task type: "xxx"`)
+	})
+
+	tests := []struct {
+		taskType         pipeline.TaskType
+		expectedTaskType interface{}
+	}{
+		{pipeline.TaskTypeHTTP, &pipeline.HTTPTask{}},
+		{pipeline.TaskTypeBridge, &pipeline.BridgeTask{}},
+		{pipeline.TaskTypeMean, &pipeline.MeanTask{}},
+		{pipeline.TaskTypeMedian, &pipeline.MedianTask{}},
+		{pipeline.TaskTypeMode, &pipeline.ModeTask{}},
+		{pipeline.TaskTypeSum, &pipeline.SumTask{}},
+		{pipeline.TaskTypeMultiply, &pipeline.MultiplyTask{}},
+		{pipeline.TaskTypeDivide, &pipeline.DivideTask{}},
+		{pipeline.TaskTypeJSONParse, &pipeline.JSONParseTask{}},
+		{pipeline.TaskTypeCBORParse, &pipeline.CBORParseTask{}},
+		{pipeline.TaskTypeAny, &pipeline.AnyTask{}},
+		{pipeline.TaskTypeVRF, &pipeline.VRFTask{}},
+		{pipeline.TaskTypeVRFV2, &pipeline.VRFTaskV2{}},
+		{pipeline.TaskTypeEstimateGasLimit, &pipeline.EstimateGasLimitTask{}},
+		{pipeline.TaskTypeETHCall, &pipeline.ETHCallTask{}},
+		{pipeline.TaskTypeETHTx, &pipeline.ETHTxTask{}},
+		{pipeline.TaskTypeETHABIEncode, &pipeline.ETHABIEncodeTask{}},
+		{pipeline.TaskTypeETHABIEncode2, &pipeline.ETHABIEncodeTask2{}},
+		{pipeline.TaskTypeETHABIDecode, &pipeline.ETHABIDecodeTask{}},
+		{pipeline.TaskTypeETHABIDecodeLog, &pipeline.ETHABIDecodeLogTask{}},
+		{pipeline.TaskTypeMerge, &pipeline.MergeTask{}},
+		{pipeline.TaskTypeLowercase, &pipeline.LowercaseTask{}},
+		{pipeline.TaskTypeUppercase, &pipeline.UppercaseTask{}},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(string(test.taskType), func(t *testing.T) {
+			t.Parallel()
+
+			taskMap := map[string]string{}
+			task, err := pipeline.UnmarshalTaskFromMap(test.taskType, taskMap, 0, "foo-dot-id")
+			require.NoError(t, err)
+			require.Equal(t, reflect.TypeOf(test.expectedTaskType), reflect.TypeOf(task))
```

Comment:
I think this helper is equivalent? :shrug: 
```suggestion
			require.IsType(t, test.expectedTaskType, task)
```

Reply:
Just did not know `.IsType`... hope this would work.